chrome/content/zotero/xpcom/zotero.js

```diff
@@ -1069,53 +1088,44 @@ Components.utils.import("resource://gre/modules/osfile.jsm");
 				fp.appendFilters(nsIFilePicker.filterAll);
 				if (fp.show() == nsIFilePicker.returnOK) {
 					var file = fp.file;
-					
-					if (file.directoryEntries.hasMoreElements()) {
+					let dialogText = '';
+					let dialogTitle = '';
+
+					// In dropbox folder
+					if (Zotero.File.isDropboxDirectory(file.path)){
+						dialogTitle = Zotero.getString('dataDir.selectedDirInDropbox.title');
+						dialogText = Zotero.getString('dataDir.selectedDirInDropbox.text');
+					}
+					else if (file.directoryEntries.hasMoreElements()) {
 						var dbfile = file.clone();
 						dbfile.append('zotero.sqlite');
 						
 						// Warn if non-empty and no zotero.sqlite
 						if (!dbfile.exists()) {
-							var buttonFlags = ps.STD_YES_NO_BUTTONS;
-							if (moreInfoCallback) {
-								buttonFlags += ps.BUTTON_POS_2 * ps.BUTTON_TITLE_IS_STRING;
-							}
-							var index = ps.confirmEx(null,
-								Zotero.getString('dataDir.selectedDirNonEmpty.title'),
-								Zotero.getString('dataDir.selectedDirNonEmpty.text'),
-								buttonFlags,
-								null,
-								null,
-								moreInfoCallback ? Zotero.getString('general.help') : null,
-								null, {});
-							
-							// Not OK -- return to file picker
-							if (index == 1) {
-								continue;
-							}
-							else if (index == 2) {
-								setTimeout(function () {
-									moreInfoCallback();
-								}, 1);
-								return false;
-							}
+							dialogTitle = Zotero.getString('dataDir.selectedDirNonEmpty.title');
+							dialogText = Zotero.getString('dataDir.selectedDirNonEmpty.text');
 						}
 					}
-					else {
+					else { // Directory empty
+						dialogTitle = Zotero.getString('dataDir.selectedDirEmpty.title');
+						dialogText = Zotero.getString('dataDir.selectedDirEmpty.text', Zotero.appName) + '\n\n'
+								+ Zotero.getString('dataDir.selectedDirEmpty.useNewDir');
+					}
+					// Warning dialog to be displayed
+					if(dialogText !== ''){
 						var buttonFlags = ps.STD_YES_NO_BUTTONS;
 						if (moreInfoCallback) {
 							buttonFlags += ps.BUTTON_POS_2 * ps.BUTTON_TITLE_IS_STRING;
 						}
 						var index = ps.confirmEx(null,
-							Zotero.getString('dataDir.selectedDirEmpty.title'),
-							Zotero.getString('dataDir.selectedDirEmpty.text', Zotero.appName) + '\n\n'
-								+ Zotero.getString('dataDir.selectedDirEmpty.useNewDir'),
+							dialogTitle,
+							dialogText,
 							buttonFlags,
 							null,
 							null,
-							moreInfoCallback ? Zotero.getString('general.moreInformation') : null,
+							moreInfoCallback ? Zotero.getString('general.help') : null,
```

Comment:
I know it was in the old code, but general.help doesn't actually exist, so this can stay general.moreInformation.